example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/compose/SessionsGooglePayScreen.kt

```diff
@@ -0,0 +1,205 @@
+                PayButton(
+                    onClick = {
+                        googlePayComponent.startGooglePayScreen(
+                            activity,
+                            SessionsGooglePayActivity.ACTIVITY_RESULT_CODE,
+                        )
+                        onButtonClicked()
+                    },
+                    allowedPaymentMethods = googlePayComponent.getGooglePayButtonParameters().allowedPaymentMethods,
+                    theme = if (useDarkTheme) ButtonTheme.Light else ButtonTheme.Dark,
+                    type = ButtonType.Pay,
+                )
+            }
+
+            SessionsGooglePayUIState.ShowComponent -> {
+                val googlePayComponent = getGooglePayComponent(componentData = googlePayState.componentData)
+                AdyenComponent(
+                    googlePayComponent,
+                    modifier,
+                )
+            }
+
+            is SessionsGooglePayUIState.FinalResult -> {
+                ResultContent(uiState.finalResult)
+            }
+        }
+    }
+}
+
+@Composable
+private fun checkGooglePayAvailability(
+    checkAvailability: GooglePayAvailabilityData?,
+    onAvailabilityChecked: () -> Unit
+): Boolean {
+    if (checkAvailability == null) return false
+    val application = LocalContext.current.applicationContext as Application
+    GooglePayComponent.PROVIDER.isAvailable(
+        application,
+        checkAvailability.paymentMethod,
+        checkAvailability.googlePayConfiguration,
+        checkAvailability.callback,
+    )
+    onAvailabilityChecked()
+    return true
+}
+
+@Composable
+private fun handleActivityResult(
+    activityResult: GooglePayActivityResult?,
+    componentData: SessionsGooglePayComponentData?,
+    onActivityResultHandled: () -> Unit
+): Boolean {
+    if (activityResult == null) return false
+    val googlePayComponent = getGooglePayComponent(componentData = componentData)
+    googlePayComponent.handleActivityResult(activityResult.resultCode, activityResult.data)
+    onActivityResultHandled()
+    return true
+}
+
+@Composable
+private fun handleAction(
+    action: Action?,
+    componentData: SessionsGooglePayComponentData?,
+    onActionConsumed: () -> Unit
+): Boolean {
+    if (action == null) return false
+    val activity = LocalContext.current as Activity
+    val googlePayComponent = getGooglePayComponent(componentData = componentData)
+    LaunchedEffect(action) {
+        googlePayComponent.handleAction(action, activity)
+        onActionConsumed()
+    }
+    return true
+}
+
+@Composable
+private fun handleNewIntent(
+    newIntent: Intent?,
+    componentData: SessionsGooglePayComponentData?,
+    onNewIntentHandled: () -> Unit
+): Boolean {
+    if (newIntent == null) return false
+    val googlePayComponent = getGooglePayComponent(componentData = componentData)
+    googlePayComponent.handleIntent(newIntent)
+    onNewIntentHandled()
```

Comment:
I would use `LaunchedEffect` here, the way it is done in `handleAction()`. That way if the state gets updated frequently, the body of the `LaunchedEffect(newIntent)` will be executed only when the value of `newIntent` changes and not on every recomposition.

Reply:
For extra safety, I would do the same for `handleActivityResult()`